commit e4e4fde51a309801af5eed72d3494cbf4b7737aa
Author: Paul Eggert <eggert@cs.ucla.edu>
Date:   Tue Sep 18 15:02:10 2018 -0700

    Fix tzfile low-memory assertion failure
    
    [BZ #21716]
    * time/tzfile.c (__tzfile_read): Check for memory exhaustion
    when registering time zone abbreviations.

diff --git a/time/tzfile.c b/time/tzfile.c
index 2a385b92bcdefec0..ea6e94030392fc75 100644
--- a/time/tzfile.c
+++ b/time/tzfile.c
@@ -410,7 +410,8 @@ __tzfile_read (const char *file, size_t extra, char **extrap)
 
   /* First "register" all timezone names.  */
   for (i = 0; i < num_types; ++i)
-    (void) __tzstring (&zone_names[types[i].idx]);
+    if (__tzstring (&zone_names[types[i].idx]) == NULL)
+      goto ret_free_transitions;
 
   /* Find the standard and daylight time offsets used by the rule file.
      We choose the offsets in the types of each flavor that are
